src/olympia/users/tests/test_admin.py

```diff
@@ -83,6 +83,150 @@ def test_search_for_multiple_user_ids(self):
+        self.client.login(email=user.email)
+        self.user.update(email='foo@bar.com', last_login_ip='127.0.0.2')
+        response = self.client.get(
+            self.list_url, {'q': '127.0.0.2,127.0.0.3'}, follow=True
+        )
+        assert response.status_code == 200
+        doc = pq(response.content)
+        # Make sure it's the right user.
+        assert doc('.field-email').text() == self.user.email
+        # Make sure last login is now displayed, and has the right value.
+        assert doc('.field-last_login_ip').text() == '127.0.0.2'
+
+    def test_search_for_multiple_ips_with_deduplication(self):
+        user = user_factory(email='someone@mozilla.com')
+        self.grant_permission(user, 'Users:Edit')
+        self.client.login(email=user.email)
+        # Will match once with the last_login
+        self.user.update(email='foo@bar.com', last_login_ip='127.0.0.2')
+        # Will match twice: once with the last login, once with the restriction history
+        # ip_address. Only one result will be shown since the 2 rows would be the same.
+        extra_user = user_factory(email='extra@bar.com', last_login_ip='127.0.0.2')
+        UserRestrictionHistory.objects.create(user=extra_user, ip_address='127.0.0.2')
+        # Will match 4 times: last_login, restriction history (last_login_ip and
+        # ip_address), ratings ip_address. There will be 2 results shown because of the
+        # 2 different user restriction history matching.
+        extra_extra_user = user_factory(
+            email='extra_extra@bar.com', last_login_ip='127.0.0.3'
+        )
+        UserRestrictionHistory.objects.create(
+            user=extra_extra_user, last_login_ip='127.0.0.2', ip_address='10.0.0.42'
+        )
+        UserRestrictionHistory.objects.create(
+            user=extra_extra_user, ip_address='127.0.0.2', last_login_ip='10.0.0.36'
+        )
+        addon = addon_factory()
+        Rating.objects.create(
+            user=extra_extra_user,
+            rating=4,
+            ip_address='127.0.0.3',
+            addon=addon,
+            version=addon.current_version,
+        )
+        response = self.client.get(
+            self.list_url, {'q': '127.0.0.2,127.0.0.3'}, follow=True
+        )
+        assert response.status_code == 200
+        doc = pq(response.content)
+        assert len(doc('#result_list tbody tr')) == 4
+        # Make sure it's the right users.
+        assert doc('.field-email').text() == ' '.join(
+            [
+                extra_extra_user.email,
+                extra_extra_user.email,
+                extra_user.email,
+                self.user.email,
+            ]
+        )
+        # Make sure last login is now displayed, and has the right values.
+        assert (
+            doc('.field-last_login_ip').text()
+            == '127.0.0.3 127.0.0.3 127.0.0.2 127.0.0.2'
+        )
+        # Same for the others that match.
+        assert (
```

Comment:
That test is unfortunately not stable, there is no guarantee in which order the "duplicates" entry for the same user (caused by multiple matching user restriction history entries) are returned. If I add an explicit `order_by()` django seems to mess with the ordering more than I'd like, need to investigate more.